google/cloud/aiplatform/models.py

```diff
@@ -578,18 +596,20 @@ def _validate_deploy_args(
         if deployed_model_display_name is not None:
             utils.validate_display_name(deployed_model_display_name)
 
-        if traffic_split is None:
-            if traffic_percentage > 100:
-                raise ValueError("Traffic percentage cannot be greater than 100.")
-            if traffic_percentage < 0:
-                raise ValueError("Traffic percentage cannot be negative.")
-
-        elif traffic_split:
-            # TODO(b/172678233) verify every referenced deployed model exists
-            if sum(traffic_split.values()) != 100:
-                raise ValueError(
-                    "Sum of all traffic within traffic split needs to be 100."
-                )
+        # TODO(b/221059294): private Endpoints do not yet support traffic splitting
+        if cls == Endpoint:
```

Comment:
Preference to refactor this so we don't make the super class need to be aware of the subclass implementation.

Reply:
Hey Sasha, sounds good. I'm going to reassign [b/221059294](https://b.corp.google.com/issues/221059294) to myself and will add the arguments to the PrivateEndpoint call and remove the line that checks if the endpoint is private or not.

Reply:
Also, I don't think we need to include any extra arguments for traffic. In the docstring for `traffic_percentage` it states `Defaults to 100 if there are no pre-existing deployed models.` and private endpoint can only have one model deployed to it as of now. Similarly, `traffic_split` has `should not be provided if traffic_percentage is provided.` I guess we can explicitly put `traffic_split=None`.

Reply:
Sasha, apologizies for not clarifiying. In the docstring for `traffic_percentage` it states: `Defaults to 100 if there are no pre-existing deployed models.`, and for `traffic_split` it states at the end: `Should not be provided if traffic_percentage is provided.` so I thought that no arguments need to be added since you values you said to pass are the default ones. I did remove exposing `Endpoint` though.

I changed
```python
        # TODO(b/221059294): private Endpoints do not yet support traffic splitting
        if cls == Endpoint:
            if traffic_split is None:
                if traffic_percentage > 100:
                    raise ValueError("Traffic percentage cannot be greater than 100.")
                if traffic_percentage < 0:
                    raise ValueError("Traffic percentage cannot be negative.")

            elif traffic_split:
                # TODO(b/172678233) verify every referenced deployed model exists
                if sum(traffic_split.values()) != 100:
                    raise ValueError(
                        "Sum of all traffic within traffic split needs to be 100."
                    )
 ```
to 
```python
        if traffic_percentage and traffic_split:
            raise ValueError(
                "Optionally define either traffic percentage or traffic split, not both."
            )
        if traffic_percentage:
            if traffic_percentage > 100:
                raise ValueError("Traffic percentage cannot be greater than 100.")
            if traffic_percentage < 0:
                raise ValueError("Traffic percentage cannot be negative.")
        if traffic_split:
            # TODO(b/172678233) verify every referenced deployed model exists
            if sum(traffic_split.values()) != 100:
                raise ValueError(
                    "Sum of all traffic within traffic split needs to be 100."
                )
```
to remove the line that checks if the endpoint is private or not and also makes sure that only one of traffic split and traffic percentage is defined.

Reply:
I'm going to go ahead and explicitly put `traffic_split=None` and `traffic_percentage=100` as well.